lib/gigo/active_record/base.rb

```diff
@@ -40,68 +40,44 @@ def after_initialize_gigoize_attributes
           private :gigoize_attributes
         end
 
-      end
-
-      module ThreeOhOnly
-
-        include Shared
-
-        def gigo_serialized_attribute(*attrs)
-          # ActiveRecord 3.0 has no proper hooks. So we fix all YAML object laoding for this class.
-          include InstanceMethods
+        def gigo_coder_for(klass)
```

Comment:
Is the dynamic class creation necessary? Seems like `GigoCoder` could accept a class to its constructor and `gigo_coder_for` could just instantiate a `GigoCoder`.

Reply:
This was my initial though process but the issue is that `serialize` accepts a `Class` and not an `Object`.

Reply:
Forget about my last comment. Turns out its possible. Updated 🙂